orchagent/isolationgrouporch.cpp

```diff
@@ -0,0 +1,754 @@
+            }
+            else
+            {
+                Port port;
+                if (!gPortsOrch->getPort(alias, port))
+                {
+                    SWSS_LOG_NOTICE("Port %s not found. Added it to m_pending_members", alias.c_str());
+                    m_pending_members.emplace_back(alias);
+                    continue;
+                }
+                addMember(port);
+            }
+        }
+        else
+        {
+            SWSS_LOG_ERROR("Port %s not supported", alias.c_str());
+            continue;
+        }
+    }
+
+    // Remove all the ports which are no longer needed
+    for (auto alias : old_members)
+    {
+        Port port;
+        if (!gPortsOrch->getPort(alias, port))
+        {
+            SWSS_LOG_ERROR("Port %s not found", alias.c_str());
+            m_pending_members.erase(find(m_pending_members.begin(), m_pending_members.end(), port.m_alias));
+        }
+        else
+        {
+            delMember(port);
+        }
+    }
+
+    return ISO_GRP_STATUS_SUCCESS;
+}
+
+isolation_group_status_t
+IsolationGroup::bind(Port &port)
+{
+    SWSS_LOG_ENTER();
+    sai_attribute_t attr;
+    sai_status_t status = SAI_STATUS_SUCCESS;
+
+    if (find(m_bind_ports.begin(), m_bind_ports.end(), port.m_alias) != m_bind_ports.end())
+    {
+        SWSS_LOG_NOTICE("isolation group %s of type %d already bound to Port %s",
+                        m_name.c_str(),
+                        m_type,
+                        port.m_alias.c_str());
+
+        return ISO_GRP_STATUS_SUCCESS;
+    }
+
+    attr.value.oid = m_oid;
+    if (m_type == ISOLATION_GROUP_TYPE_BRIDGE_PORT)
+    {
+        if (port.m_bridge_port_id != SAI_NULL_OBJECT_ID)
+        {
+            attr.id = SAI_BRIDGE_PORT_ATTR_ISOLATION_GROUP;
+            status = sai_bridge_api->set_bridge_port_attribute(port.m_bridge_port_id, &attr);
+            if (SAI_STATUS_SUCCESS != status)
+            {
+                SWSS_LOG_ERROR("Unable to set attribute %d value  0x%" PRIx64 "to %s",
+                               attr.id,
+                               attr.value.oid,
+                               port.m_alias.c_str());
+            }
+            else
+            {
+                m_bind_ports.push_back(port.m_alias);
+            }
+        }
+        else
+        {
+            m_pending_bind_ports.push_back(port.m_alias);
+            SWSS_LOG_NOTICE("Port %s saved in pending bind ports for isolation group %s of type %d",
+                            port.m_alias.c_str(),
+                            m_name.c_str(),
+                            m_type);
+        }
+    }
+    else if (m_type == ISOLATION_GROUP_TYPE_PORT)
+    {
+        if ((port.m_type == Port::PHY ? port.m_port_id : port.m_lag_id) != SAI_NULL_OBJECT_ID)
+        {
+            attr.id = SAI_PORT_ATTR_ISOLATION_GROUP;
+            status = sai_port_api->set_port_attribute((port.m_type == Port::PHY ? port.m_port_id : port.m_lag_id), &attr);
```

Comment:
Please correct me if I'm wrong but I think `port.m_lag_id` contains LAG OID. So, should not be used to set port's attribute
https://github.com/Azure/sonic-swss/blob/1bc94d15281cc294bf07dffe30905408133b7559/orchagent/portsorch.cpp#L1186
Same in other places.

Reply:
Btw, `::unbind()` uses `port.m_port_id` only. So, looks like one of these two is wrong..

Reply:
@akokhan  Thanks for the comment. The LAG support is added considerint future support. We are removing the LAG specific code for now as the support is not there.

Reply:
What do you mean by the LAG support is not there? I thought this PR is about implementing MCLAG support.

Maybe you can clarify: Will SONiC have MCLAG support after this PR is merged?

Reply:
@haslersn: the question here is specific to isolation group change. At present there is no SAI support for LAG level ISO gourp, its only done at member PORT level.
Yes with this PR MCLAG is supported over PortChannel.  We are working on fixing some of the pending VS tests to make it ready for push.